aspnetcore/migration/1x-to-2x/identity-2x.md

```diff
@@ -0,0 +1,272 @@
+keywords: ASP.NET Core,Identity,authentication
+ms.author: scaddie
+manager: wpickett
+ms.date: 08/01/2017
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: migration/identity-2x
+---
+
+# Migrating Authentication and Identity to ASP.NET Core 2.0
+
+By [Scott Addie](https://github.com/scottaddie) and [Hao Kung](https://github.com/HaoK)
+
+ASP.NET Core 2.0 has a new model for authentication and [Identity](xref:security/authentication/identity) which simplifies configuration by using services. ASP.NET Core 1.x applications that use authentication or Identity can be updated to use the new model as outlined below.
+
+<a name="auth-middleware"></a>
+
+## Authentication Middleware and Services
+In 1.x projects, authentication is configured via middleware. A middleware method is invoked for each authentication scheme you want to support.
+
+The following 1.x example configures Facebook authentication with Identity in *Startup.cs*:
+
+```csharp
+public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory) {
+    app.UseIdentity();
+    app.UseFacebookAuthentication(new FacebookOptions { 
+        AppId = Configuration["auth:facebook:appid"],
+        AppSecret = Configuration["auth:facebook:appsecret"]
+    });
+} 
+```
+
+In 2.0 projects, authentication is configured via services. Each authentication scheme is registered in the `ConfigureServices` method of *Startup.cs*. The `UseIdentity` method is replaced with `UseAuthentication`.
+
+The following 2.0 example configures Facebook authentication with Identity in *Startup.cs*:
+
+```csharp
+public void ConfigureServices(IServiceCollection services) {
+    services.AddIdentity<ApplicationUser, IdentityRole>().AddEntityFrameworkStores();
+
+    // If you want to tweak Identity cookies, they're no longer part of IdentityOptions.
+    services.ConfigureApplicationCookie(o => o.LoginPath = "/Account/LogIn");
+    services.AddAuthentication()
+            .AddFacebook(o => {
+                o.AppId = Configuration["facebook:appid"];
+                o.AppSecret = Configuration["facebook:appsecret"];
+            });
+}
+
+public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory) {
+    app.UseAuthentication();
+}
+```
+
+The `UseAuthentication` method discovers the services being used, from [dependency injection](xref:fundamentals/dependency-injection), to determine the authentication schemes you want to use.
+
+Below are 2.0 migration instructions for each major authentication scheme.
+
+### Cookie-based Authentication
+Select one of the two options below, and make the necessary changes in *Startup.cs*:
+
+1. Use cookies with Identity
+    - Invoke the `AddIdentity` method in the `ConfigureServices` method to add the cookie authentication services.
+    - Invoke the `ConfigureApplicationCookie` or `ConfigureExternalCookie` method in the `ConfigureServices` method to configure the Identity cookie.
```

Comment:
No the other two factor cookies don't have sugar methods, you'd have to configure those via named options and the constants, its very rare that people need to tweak those, those are more internal cookies than anything else